### ISSUE TITLE
PyData Paris 2017

### ISSUE THREAD

**Issue by jfbercher** [ipython-contrib/jupyter_contrib_nbextensions#964]
I will have the opportunity to go to [pydata Paris 2017](http://pyparis.org/). Following @janschulz in #959, it could be a good idea to submit there something on the extensions developped by the group and/or collected in this repo? (_deadline of the CFP on April 23_)

Below is a tentative proposal (with what I know the best). Any thoughts/reactions on that? Would you agree with that, particulary @juhasch , @jcb91 @janschulz  ?  How should I list the authors? (I don't think that there is a published paper after, though)

# Some extensions for Jupyter/IPython notebook

This talk will present and describe some Jupyter/IPython notebook extensions by the author(s) and others of the [jupyter_contrib_nbextensions](https://github.com/ipython-contrib/jupyter_contrib_nbextensions) group which are useful for 
- core development and data analysis,
- publication of research reports and papers,
- classroom animation.

The Jupyter (formerly IPython) notebook is a browser application that allows to create and share live documents which embed text, code and multimedia. This enables to use the notebook as an IDE with integrated documentation capabilities, to use it for reproductible research as well as an authoring system, or to prepare demos or full experimentations for teaching. In these three areas (and more), the basic capabilities of the notebook can be extended through notebook extensions. 

We will describe the general extension structure and the available API, as well as the main ideas, structures and practices that have been used in the last extensions submitted to [jupyter_contrib_nbextensions](https://github.com/ipython-contrib/jupyter_contrib_nbextensions). Several extensions will be presented, such as
- code developpement: [code_prettify](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/code_prettify/README_code_prettify.html) (code formatter via yapf or auto PEP8), [2to3](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/code_prettify/README_2to3.html)  (python converter), [runtools](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/runtools/readme.html), [varinspector](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/varInspector/README.html) (variable inspector), 
- notebook use and publication: [toc2](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/toc2/README.html) (table of contents/navigation menu), [collapsible headings](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/collapsible_headings/readme.html), [python markdown](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/python-markdown/readme.html), [hinterland](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/hinterland/README.html), [nbTranslate](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/nbTranslate/README.html) (multilanguage support), [latex_envs](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/latex_envs/README.html) (use LaTeX environments within the notebook -- and export them)
- classroom: [exercise](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/exercise/readme.html), [exercise2](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/exercise2/readme.html) (include exercises with hidden solutions), [highlighter](http://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/highlighter/readme.html) (select, highlight, annotate text). 

Implementation details will be given for some of these extensions.

**Comment by jcb91** [comment]
Neat :smiley: yes, it seems like a nice idea to me! I quite like [jupyter_highlight_selected_word](https://github.com/jcb91/jupyter_highlight_selected_word), notify and execute_time for development work/usage, and I think it might be worth mentioning codefolding and select_keymap for usability, and possibly snippets_menu for classroom use (an educator might provide a set of snippets tailored to a class?).

As for authors, I wouldn't worry too much, really, so long as the repo is listed. I think on the abstract, probably just list the repo & maybe `IPython-contrib/Jupyter-contrib developers`, as otherwise it'll be a very long list which is probably a a pain to extract :laughing: It could work to give authors for individual extensions in the talk, but again, it'd need (for me at least) a check of the commit history to remember completely who's worked on which ones! But ultimately, you can probably judge what's appropriate for the conference better than I can :smile:

**Comment by janschulz** [comment]
Is someone going to pydata Berlin? I will probably be there, maybe submit the talk there as well?

**Comment by jfbercher** [comment]
Yes, of course for submitting there as well! I have submitted a proposal updated with @jcb91 's comments. 
If it is selected, I will have to prepare some slides and demos and we will be able to share that (and obviously inputs/contribs will be welcome).

**Comment by janschulz** [comment]
If I go to pydata berlin, I would like to submit a lightening talk (it's organized during the conference, so no need to submit a proposal yet). Having some slides already would be great :-)

**Comment by juhasch** [comment]
@jfbercher: Good idea. I think codefolding and skip-traceback with recent update from @jcb91 might be interesting, too. 
@janschulz: I won't go to pydata Berlin, but probably to EuroScipy in Erlangen.

I will give a talk on extensions soon, too. Maybe we could create a new repo with example notebooks, so we can reuse this in our presentations.
